### ISSUE TITLE
Demon's Souls does not function properly with mods on NA/EU versions

### ISSUE THREAD

**Issue by beanrouth** [RPCS3/rpcs3#9082]
I'm not sure if this is an issue with the emulator exactly, because Demon's Souls functions perfectly on the emulator without mods. It's when mods are introduced that problems occur. I've tried for hours to fix this, and I really don't know where else to ask for help.

The problem: After installing mods on Demon's Souls, loading screens are preceded by a black screen that lasts for a few seconds. Not only that, but on startup, the Demon's Souls screen will get stuck for a minute at a time on a screen that usually lasts two seconds. Sometimes, the main menu doesn't load from this screen at all. This is not my screenshot, but here's an image of the screen I'm talking about: https://user-images.githubusercontent.com/11798478/45638107-56649480-bad6-11e8-9ac6-5613359d7b1e.png

Otherwise, I've ran into more problems, but they don't appear as often. Sometimes the game freezes on loading screens and won't progress further, and I've also had audio cut out and repeat itself, even after I've quit out to the main menu. This doesn't happen if mods aren't installed.

From what I can find, it seems like some people are having these issues, but not everyone. I've tried on two different PC's with the same result. The game simply does not function properly with mods. I've even performed a clean install of Windows 10 on my main PC, because it's fairly new and I didn't need to reinstall a whole lot.

The three mods I'm attempting to use are the lens flare fix, gray outline removal, and high resolution font. Installation of the mods is very simple, so I'm positive I'm not messing that up. The only confusing part is that some of them require you to delete the "cache" folder under dev_hdd1. This folder doesn't exist for me, so I'm assuming that's from a previous build of RPCS3. What does exist is a "caches" folder. I've deleted that and also tried to other solutions: Open RPCS3>Right Click Demon's Souls>Remove All Caches & File>All Titles>Remove Disc Cache.

I've tried previous builds of RPCS3 that people claim to work. They have not solved my issues.

This appears on either disc version of the game, EU or NA (BLES00932/BLUS30443). I started with a copy of the EU version that I downloaded online, then I decided to rip the files straight from my NA copy of Demon's Souls after I wasn't getting anywhere with the EU version. This also happens at 30 FPS or with the 60 FPS patch enabled.

Solutions attempted:
- Download on 2 different PCs
- Clean install Windows 10
- Uninstall and reinstall disc copies of Demon's Souls
- Uninstall and reinstall mods
- Delete different caches after installing mods
- Boot games for the first time with mods already installed, so cache deletion wouldn't even be necessary

These mods are not a MUST, but they improve the experience greatly. I would really appreciate any guidance on how to solve this. 

Please see my log file: https://mega.nz/file/rF03SIbB#n8WEuSzKXHqBI_d42BG4uDsDoR43uc4i-TX-jU_MZwI

PC Specs:
OS: Windows 10 Home
CPU: Intel i7-7820X
GPU: Geforce RTX 2080
RAM: GSKILL Ripjaws V 16GB DDR4 3200MHz
MB: ASUS TUF x299 MARK 2 ATX
SSD: 500GB WD Blue Series
HDD: 3TB 7200 RPM
PSU: Corsair TX850M 850W

**Comment by HerrHulaHoop** [comment]
> The three mods I'm attempting to use are the lens flare fix, gray outline removal, and high resolution font

None of these patches are from our Wiki. You've got them from some other source. If the game works without the patches, then the issue is most likely present in the patches themselves. Game patches are made by the community so you need to reach out to the author of the patches to have this issue addressed. RPCS3 can't do much about this.

**Comment by mshpakov** [comment]
"gray outline removal"
"high resolution font"
"Boletaria Reborn v2.1" (including lens flare fix)

These mods are working quite well on both BLES00932 and BLUS30443 versions.
Completely no issues with latest RPCS builds and 4k/ 60  fps patches enabled.

Looks like your game copy is not vanilla or you've messed with your RPCS settings for the game.

**Comment by beanrouth** [comment]
> > The three mods I'm attempting to use are the lens flare fix, gray outline removal, and high resolution font
> 
> None of these patches are from our Wiki. You've got them from some other source. If the game works without the patches, then the issue is most likely present in the patches themselves. Game patches are made by the community so you need to reach out to the author of the patches to have this issue addressed. RPCS3 can't do much about this.

Yeah I understand what you’re saying, I just wanted to give it a shot and see if anyone here can help. Thanks!

**Comment by beanrouth** [comment]
> "gray outline removal"
> "high resolution font"
> "Boletaria Reborn v2.1" (including lens flare fix)
> 
> These mods are working quite well on both BLES00932 and BLUS30443 versions.
> Completely no issues with latest RPCS builds and 4k/ 60 fps patches enabled.
> 
> Looks like your game copy is not vanilla or you've messed with your RPCS settings for the game.

Like I said in the post, I know these mods work well for probably most people. Not everyone runs into the issues that I am. Some people do if you read the posts on the mods though. I can’t vouch for the EU version I downloaded, but my US version is ripped straight from my disc copy of DeS. It’s 100% vanilla. Last night I placed each mod in my USRDIR folder one by one, and realized exactly what each mod is doing. The lens flare mod actually works fine. The Font mod is what causes the game to take forever to get to the main menu, and the Gray Outline mod is what causes the longer loading screens. I know I’m not installing the mods incorrectly. At this point though, I give up, because obviously I’m in the minority with these problems.

**Comment by mshpakov** [comment]
Check settings for your DS game in RPCS3. 
My custom settings for this game are: 

Renderer: Vulkan
Resolution scale: 400%
Write color buffers: On
Multithreaded RSX: On
Audio buffer: 150 ms

All other settings are default.

Also don't forget to remove all caches for the game, you can do it from GUI , just right click on game title -> Remove -> Remove All Caches

**Comment by beanrouth** [comment]
> Check settings for your DS game in RPCS3.
> My custom settings for this game are:
> 
> Renderer: Vulkan
> Resolution scale: 400%
> Write color buffers: On
> Multithreaded RSX: On
> Audio buffer: 150 ms
> 
> All other settings are default.
> 
> Also don't forget to enable 60fps/aspect ratio patches and remove all caches for the game before you start it. You can do it from GUI , just right click on game title -> Remove -> Remove All Caches

Thanks! I actually came up with a solution I'm satisfied with. Kinda hard to explain. So like I said, I realized the lens flare mod was good and didn't cause any problems. Then when you extract the Gray Outline mod, there's two folders: menu and other. I figured out that the "other" folder didn't cause any problems with the game either. I downloaded programs to view all of the files in the menu folder that the modded menu folder also had. Some of the files were the actual loading screens themselves, like the pictures of Executioner Miralda and Mephistopheles while warping. The other files were for things like the menu and stamina/health bars. I copied all of the files that had to do with the menu and OSD, and excluded the loading screen files. The game works perfectly fine. Not sure why, but I guess there's an issue with loading the modded loading screen images, and that's what causes the black screen delay before they appear. I don't really care about the lines on the loading screens, just the in-game outlines. So all is good. I'm not even messing with the high res font mod, I don't care about that one. This is absolutely not a problem with RPCS3, but still useful info for anyone else this might happen to. 👍
